cmd/opm/serve/serve.go

```diff
@@ -59,12 +68,32 @@ will not be reflected in the served content.
 	}
 
 	cmd.Flags().BoolVar(&s.debug, "debug", false, "enable debug logging")
+	cmd.Flags().StringVar(&s.profilePort, "profileport", "", "pprof HTTP endpoint port")
+	cmd.Flags().StringVar(&s.cpuProfile, "cpuprofile", "", "output file for CPU profiling data dump")
+	cmd.Flags().StringVar(&s.memProfile, "memprofile", "", "output file for memory profiling data dump")
 	cmd.Flags().StringVarP(&s.port, "port", "p", "50051", "port number to serve on")
 	cmd.Flags().StringVarP(&s.terminationLog, "termination-log", "t", "/dev/termination-log", "path to a container termination log file")
 	return cmd
 }
 
 func (s *serve) run(ctx context.Context) error {
+	if s.profilePort != "" {
+		go func() {
+			s.logger.Info(http.ListenAndServe("localhost:"+s.profilePort, nil))
+		}()
+	}
+	if s.cpuProfile != "" {
+		f, err := os.Create(s.cpuProfile)
```

Comment:
Should we sanitize the user-input somehow before processing it?

Reply:
We should, so at least for this PR we're just going to use well-known/communicated output locations, so that we don't have to.